tfhe-zk-pok/src/proofs/pke_v2.rs

```diff
@@ -328,8 +328,14 @@ pub fn prove<G: Curve>(
 
     let PrivateCommit { r, e1, m, e2, .. } = private_commit;
 
-    assert!(c2.len() <= k);
-    let k = k.min(c2.len());
+    assert!(c2.len() <= k_max);
+    let k = k_max.min(c2.len());
+
+    // Recompute the D for our case if k is smaller than the k max
+    // formula in Prove_pp: 2.
+    let D = d + k * t_input.ilog2() as usize;
+
+    assert!(D <= D_max);
```

Comment:
this API use a result on one side and asserts on the other side, this could be uniformized (but that's maybe out of the scope of this pr)